indexer/client/client.go

```diff
@@ -0,0 +1,164 @@
+package client
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"encoding/json"
+
+	"github.com/ethereum-optimism/optimism/indexer/api"
+	"github.com/ethereum-optimism/optimism/indexer/database"
+	"github.com/ethereum-optimism/optimism/indexer/node"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	urlParams = "?cursor=%s&limit=%d"
+
+	defaultPagingLimit = 100
+)
+
+// Config ... Indexer client config struct
+type Config struct {
+	PaginationLimit int
+	URL             string
```

Comment:
nit I think more clear if this is named baseurl

Reply:
Agreed: [75c6231](https://github.com/ethereum-optimism/optimism/pull/7312/commits/75c623151365909c672ed4e1624257caa912eb19)